src/App.tsx

```diff
@@ -1,51 +1,89 @@
-import React from 'react';
+import classNames from 'classnames';
+import { Component } from 'react';
 import './App.scss';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type State = {
+  selectedGood: string,
 };
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+    const isSelectedGood = selectedGood
+      ? `${selectedGood} is selected`
+      : 'No goods selected';
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+            {isSelectedGood}
+          </h1>
+
+          <button
+            hidden={!selectedGood}
+            type="button"
+            className={classNames(
+              'App__clear', 'button',
+              {
+                button__hidden: !selectedGood,
+              },
+            )}
+            onClick={() => {
+              this.setState({ selectedGood: '' });
+            }}
+          >
+            Clear
+          </button>
+        </header>
+
+        <ul className="App__list">
+          {goodsFromServer.map(good => (
+            <li
+              key={good}
+              className={classNames(
+                'Good',
+                { 'Good--active': good === selectedGood },
+              )}
+            >
+              {good}
+              <button
+                type="button"
+                className={classNames(
+                  'Good', 'button', 'Good__remove',
+                  { button__hidden: good !== selectedGood },
```

Comment:
I guess hidden is'nt a BEM element but a modifier so 
```suggestion
                  { button--hidden: good !== selectedGood },
```